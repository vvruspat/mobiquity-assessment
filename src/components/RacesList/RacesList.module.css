.list {
    display: flex;
    justify-content: flex-start;
    flex-direction: column;
    gap: var(--padding-medium);
    margin-top: var(--padding-medium);
}