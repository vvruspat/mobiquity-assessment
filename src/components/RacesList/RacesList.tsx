import { RaceWinner } from "../../types/winner";
import { RaceCard } from "../RaceCard";
import { AnimatedList } from "react-animated-list";

import s from "./RacesList.module.css";

type RacesListProps = {
  winners: RaceWinner[];
  winner: RaceWinner;
};

/** List of races winners */
export const RacesList = ({ winners, winner }: RacesListProps) => {
  return (
    <div className={s.list}>
      <AnimatedList animation="grow" initialAnimationDuration={5000}>
        {winners.map((currentWinner) => (
          <RaceCard
            {...currentWinner}
            key={currentWinner.round}
            isAccent={currentWinner.driver === winner.driver}
          />
        ))}
      </AnimatedList>
    </div>
  );
};
