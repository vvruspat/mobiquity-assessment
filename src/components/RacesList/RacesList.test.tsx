import { render, screen } from "@testing-library/react";
import { RacesList } from "./RacesList";

const props = {
  winner: {
    round: "2",
    date: "1979-10-17",
    raceTitle: "Spanish Grand Prix",
    country: "Spain",
    circuit: "Circuit de Barcelona-Catalunya",
    driver: "Charles Leclerc",
    points: 256,
    time: "12:34:34.765",
    speed: "256.65",
    team: "RED BULL",
    nationality: "Spanish",
  },
  winners: [
    {
      round: "1",
      date: "1979-10-17",
      raceTitle: "Spanish Grand Prix",
      country: "Spain",
      circuit: "Circuit de Barcelona-Catalunya",
      driver: "Charles Leclerc",
      points: 256,
      time: "12:34:34.765",
      speed: "256.65",
      team: "BMW",
      nationality: "Dutch",
    },
    {
      round: "2",
      date: "1979-10-17",
      raceTitle: "Spanish Grand Prix",
      country: "Spain",
      circuit: "Circuit de Barcelona-Catalunya",
      driver: "Charles Leclerc",
      points: 256,
      time: "12:34:34.765",
      speed: "256.65",
      team: "RED BULL",
      nationality: "Spanish",
    },
  ],
};

test("renders races list", () => {
  render(<RacesList {...props} />);

  const el = screen.getByTestId("races-list");
  expect(el).toMatchSnapshot();

  const cards = screen.getAllByTestId("card");
  expect(cards.length).toBe(2);
});
