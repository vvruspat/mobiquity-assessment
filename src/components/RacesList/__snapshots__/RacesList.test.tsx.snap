// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders races list 1`] = `
<div
  class="list"
  data-testid="races-list"
>
  <div
    style="opacity: 1; transform: scale(1, 1); transition: opacity 2500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 1665ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    <section
      class="card card_accent"
      data-testid="card"
    >
      <div
        class="content"
      >
        <header
          class="header"
        >
          <div
            class="header_info"
          >
            <div
              class="round"
            >
              <div
                class="badge badge_s badge_style_accent badge_align_center round_badge"
                data-testid="badge"
              >
                <span
                  class="text size_s regular"
                  data-testid="text"
                >
                  Round 
                  1
                </span>
                <span
                  class="v_separator"
                >
                  |
                </span>
                <span
                  class="text size_s regular"
                  data-testid="text"
                >
                  October 17, 1979
                </span>
              </div>
            </div>
            <h2
              class="title"
              data-testid="title"
            >
              <span
                class="text size_xl bold"
                data-testid="text"
              >
                Spanish Grand Prix
              </span>
            </h2>
            <h6
              class="caption hidden"
              data-testid="caption"
            >
              <span
                class="text size_s regular"
                data-testid="text"
              >
                Circuit de Barcelona-Catalunya
              </span>
            </h6>
          </div>
          <div
            class="headerControls"
          >
            <button
              class="button icon-button"
              data-testid="button"
            >
              <svg>
                arrow.svg
              </svg>
            </button>
          </div>
        </header>
        <hr
          class="separator hidden"
          data-testid="separator"
        />
        <section
          class="content content_collapsed"
        >
          <div
            class="flag"
          >
            <div
              class="flag flag_s fi fi-nl"
              data-testid="flag"
            />
          </div>
          <div
            class="info info_collapsed"
          >
            <div
              class="winner hidden"
            >
              <h6
                class="caption"
                data-testid="caption"
              >
                <span
                  class="text size_s regular"
                  data-testid="text"
                >
                  Winner
                </span>
              </h6>
            </div>
            <div
              class="driver"
            >
              <span
                class="text size_l semibold"
                data-testid="text"
              >
                Charles Leclerc
              </span>
            </div>
          </div>
          <div
            class="stat"
          >
            <div
              class="badge badge_m badge_style_primary badge_align_center"
              data-testid="badge"
            >
              <div
                class="team"
              >
                <span
                  class="text size_s regular"
                  data-testid="text"
                >
                  BMW
                </span>
              </div>
            </div>
          </div>
        </section>
        <footer
          class="footer hidden"
        >
          <div
            class="badge badge_l badge_style_info badge_align_left"
            data-testid="badge"
          >
            <h6
              class="caption"
              data-testid="caption"
            >
              <span
                class="text size_s regular"
                data-testid="text"
              >
                Time
              </span>
            </h6>
            <span
              class="text size_m regular"
              data-testid="text"
            >
              12:34:34.765
            </span>
          </div>
          <div
            class="badge badge_l badge_style_info badge_align_left"
            data-testid="badge"
          >
            <h6
              class="caption"
              data-testid="caption"
            >
              <span
                class="text size_s regular"
                data-testid="text"
              >
                Km/h
              </span>
            </h6>
            <span
              class="text size_m regular"
              data-testid="text"
            >
              256.65
            </span>
          </div>
          <div
            class="badge badge_l badge_style_info badge_align_left"
            data-testid="badge"
          >
            <h6
              class="caption"
              data-testid="caption"
            >
              <span
                class="text size_s regular"
                data-testid="text"
              >
                Points
              </span>
            </h6>
            <span
              class="text size_m regular"
              data-testid="text"
            >
              256
            </span>
          </div>
        </footer>
      </div>
    </section>
  </div>
  <div
    style="opacity: 1; transform: scale(1, 1); transition: opacity 5000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 3330ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    <section
      class="card card_accent"
      data-testid="card"
    >
      <div
        class="content"
      >
        <header
          class="header"
        >
          <div
            class="header_info"
          >
            <div
              class="round"
            >
              <div
                class="badge badge_s badge_style_accent badge_align_center round_badge"
                data-testid="badge"
              >
                <span
                  class="text size_s regular"
                  data-testid="text"
                >
                  Round 
                  2
                </span>
                <span
                  class="v_separator"
                >
                  |
                </span>
                <span
                  class="text size_s regular"
                  data-testid="text"
                >
                  October 17, 1979
                </span>
              </div>
            </div>
            <h2
              class="title"
              data-testid="title"
            >
              <span
                class="text size_xl bold"
                data-testid="text"
              >
                Spanish Grand Prix
              </span>
            </h2>
            <h6
              class="caption hidden"
              data-testid="caption"
            >
              <span
                class="text size_s regular"
                data-testid="text"
              >
                Circuit de Barcelona-Catalunya
              </span>
            </h6>
          </div>
          <div
            class="headerControls"
          >
            <button
              class="button icon-button"
              data-testid="button"
            >
              <svg>
                arrow.svg
              </svg>
            </button>
          </div>
        </header>
        <hr
          class="separator hidden"
          data-testid="separator"
        />
        <section
          class="content content_collapsed"
        >
          <div
            class="flag"
          >
            <div
              class="flag flag_s fi fi-es"
              data-testid="flag"
            />
          </div>
          <div
            class="info info_collapsed"
          >
            <div
              class="winner hidden"
            >
              <h6
                class="caption"
                data-testid="caption"
              >
                <span
                  class="text size_s regular"
                  data-testid="text"
                >
                  Winner
                </span>
              </h6>
            </div>
            <div
              class="driver"
            >
              <span
                class="text size_l semibold"
                data-testid="text"
              >
                Charles Leclerc
              </span>
            </div>
          </div>
          <div
            class="stat"
          >
            <div
              class="badge badge_m badge_style_primary badge_align_center"
              data-testid="badge"
            >
              <div
                class="team"
              >
                <span
                  class="text size_s regular"
                  data-testid="text"
                >
                  RED BULL
                </span>
              </div>
            </div>
          </div>
        </section>
        <footer
          class="footer hidden"
        >
          <div
            class="badge badge_l badge_style_info badge_align_left"
            data-testid="badge"
          >
            <h6
              class="caption"
              data-testid="caption"
            >
              <span
                class="text size_s regular"
                data-testid="text"
              >
                Time
              </span>
            </h6>
            <span
              class="text size_m regular"
              data-testid="text"
            >
              12:34:34.765
            </span>
          </div>
          <div
            class="badge badge_l badge_style_info badge_align_left"
            data-testid="badge"
          >
            <h6
              class="caption"
              data-testid="caption"
            >
              <span
                class="text size_s regular"
                data-testid="text"
              >
                Km/h
              </span>
            </h6>
            <span
              class="text size_m regular"
              data-testid="text"
            >
              256.65
            </span>
          </div>
          <div
            class="badge badge_l badge_style_info badge_align_left"
            data-testid="badge"
          >
            <h6
              class="caption"
              data-testid="caption"
            >
              <span
                class="text size_s regular"
                data-testid="text"
              >
                Points
              </span>
            </h6>
            <span
              class="text size_m regular"
              data-testid="text"
            >
              256
            </span>
          </div>
        </footer>
      </div>
    </section>
  </div>
</div>
`;
